amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/Lease.java

```diff
@@ -91,6 +91,16 @@
      */
     private byte[] pendingCheckpointState;
 
+
+    /**
+     * Denotes whether the lease is marked for stealing. Deliberately excluded from hashCode and equals and
+     * not persisted in DynamoDB.
+     *
+     * @return flag for denoting lease is marked for stealing.
+     */
+    @Setter
+    private boolean isMarkedForLeaseSteal;
```

Comment:
I can create a `LeaseWithMetadata` wrapper that takes in a `Lease` and `isMarkedForLeaseSteal`, but that requires a bit more work, which will make this PR a bit bigger since it requires replacing instances of `Lease` with `LeaseWithMetadata`

I can submit a follow up PR with that change